app/controllers/concerns/voted.rb

```diff
@@ -0,0 +1,25 @@
+module Voted
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_votable, only: [:like, :unlike]
+  end
+
+  def like
+    instance_variable_set("@#{controller_name.singularize}", @votable).liked_by current_user if !(current_user.author?(instance_variable_set("@#{controller_name.singularize}", @votable)))
```

Comment:
We can move this code to `set_votable`, can't we?
And then simply use `@votable` in actions.